core/chains/evm/client/client.go

```diff
@@ -202,6 +189,11 @@ func (client *client) SendTransaction(ctx context.Context, tx *types.Transaction
 	return client.pool.SendTransaction(ctx, tx)
 }
 
+func (client *client) SimulateTransaction(ctx context.Context, tx *types.Transaction) error {
```

Comment:
What is the functionality of `SimulateTransaction` on other chains? Is there a link with `EstimateGasPrice`?

Reply:
`SimulateTransaction` allows to send signed/unsigned tx to the node and check it's validity against the current state. It's an easier and more lightweight way comparing to signing the transaction, estimating fees etc for the real tx and then parsing the revert errors. Also on some chains (e.g. Starknet) it's combined with the final fee estimation.